bootloader/piboot.go

```diff
@@ -375,7 +383,55 @@ func (p *piboot) layoutKernelAssetsToDir(snapf snap.Container, dstDir string) er
 	return nil
 }
 
+func (p *piboot) eepromVersionSupportsTryboot() bool {
+	// To find out the EEPROM version we do the same as the
+	// rpi-eeprom-update script (see
+	// https://github.com/raspberrypi/rpi-eeprom/blob/master/rpi-eeprom-update)
+	buf, err := ioutil.ReadFile(rpi4EepromTimeStampPath)
+	if err != nil {
+		return false
+	}
+
+	// The timestamp is seconds since the epoch, UTC time
+	eepromTs := binary.BigEndian.Uint32(buf)
+	// 2021-03-18 or more modern supports tryboot, see
+	// https://github.com/raspberrypi/rpi-eeprom/blob/master/firmware/release-notes.md#2021-04-19---promote-2021-03-18-from-latest-to-default---default
+	// The timestamp we compare with (jue 18 mar 2021 08:54:11 UTC) can be found with:
+	// $ strings pieeprom-2021-03-18.bin | grep BUILD_TIMESTAMP
+	if eepromTs < 1616057651 {
+		return false
+	}
+
+	return true
+}
+
+func (p *piboot) isRaspberryPi4() bool {
+	// For RPi4 detection we do the same as the rpi-eeprom-update script (see
+	// https://github.com/raspberrypi/rpi-eeprom/blob/master/rpi-eeprom-update)
+	buf, err := ioutil.ReadFile(rpi4RevisionCodesPath)
+	if err != nil {
+		return false
+	}
+
+	// If old style codes (older than RPi2) or processor != BCM2711 (RPi4's SoC),
+	// this is not an RPi4. For details, see
+	// https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-revision-codes
+	boardInfo := binary.BigEndian.Uint32(buf)
+	if ((boardInfo>>23)&1) == 0 || ((boardInfo>>12)&0xF) != 3 {
+		return false
+	}
+
+	return true
+}
+
 func (p *piboot) ExtractKernelAssets(s snap.PlaceInfo, snapf snap.Container) error {
+	if !p.prepareImageTime {
+		// If this is a RPi4, check first if EEPROM supports tryboot
+		if p.isRaspberryPi4() && !p.eepromVersionSupportsTryboot() {
+			return fmt.Errorf("your EEPROM does not support tryboot, please upgrade to a newer one before installing Ubuntu Core")
```

Comment:
One nice touch would be if we could link to some forum topic with some more help. E.g. 
```
your EEPROM does not support tryboot, please upgrade to a newer one before installing Ubuntu Core, see http://forum.snapcraft.io/t/29455 for more details
```
(I created this unlisted page that probably needs some tweaks)

Reply:
Nice idea, I've changed the message accordingly.